### ISSUE TITLE
Accordion toggles in the Connectors > AI plugin page don't follow the APG pattern and miss a focus style

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#82986]
### Description

The AI plugin page in the  WP admin contains accordions.

These accordions don't follow the ARIA Authoring Practices Gyude (APG) design patterns.
Also, the accordion toggles don't have any focus style

[According to the APG](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/), accordions should:
- Use headings that wrap a button element.
- The button is the actual trigger to toggle the accordion.
- The headings identify the section of content within the accotdions.

See also the APG [example implementation](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/).

Instead, in the AI plugin page:
- The accordion toggle are clickable div elements with a role=button.
- The toggles don't use any heading.
- There is no focus style at all.

<img width="1570" height="750" alt="Image" src="https://github.com/user-attachments/assets/73d029ec-eec6-4c9e-a21b-caabf72c8c18" />

### Step-by-step reproduction instructions

- Go to WP Admin > Connectors.
- Install and activate the AI plugin.
- You will be redirected to the AI plugin settings page `wp-admin/options-general.php?page=ai-wp-admin`
- Use the tab key to navigate the page.
- Onserve that when you land on on one of the accordion toggles (Editor Experiments or Admin Experiments) there is a tab stop but there is no focus style.
- Inspect the DOM and observe the accortdion toggles are div elements with a role=button.
- Observe they don't contain any heading.

Regarding clickable div elements, I though we made clear years ago that they should not be used. Just use buttons. Modern browsers allow to fully style buttons. There is no reason to use clickable div elements and replicate via JavaScript all native functions of button elements.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by mirka** [comment]
> * The accordion toggle are clickable div elements with a role=button.

For context, is there anything missing in the implementation that should be available with a native button? I think everything should be polyfilled by the upstream library. Mostly asking because I'm not sure we can actually use a native button here, given the use cases and composition patterns. We'd have to check.

> * The toggles don't use any heading.

I believe these are intended to be a [disclosure](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/) pattern (not accordion), so we need to evaluate them from that perspective. But if headings or other semantics are appropriate in certain contexts (thinking about this [disclosure card](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-card/) for example), that may require additional APIs to `DataForm` (cc @oandregal).

> * There is no focus style at all.

The underlying components (`CollapsibleCard`, `DataForm`) are showing the focus rings, so that's likely an issue for the [AI plugin repo](https://github.com/WordPress/ai). I think they may be using an older version of `DataForm` that doesn't include the fix in #77468.

**Comment by hbhalodia** [comment]
> There is no focus style at all.

Hi @afercia @mirka, I have created an issue over AI repo and raised the PR with an update to latest version of `@wordpress/dataviews` pacakge, which proves to fix the `focus` issue.

Issue - https://github.com/WordPress/ai/issues/1034
PR - https://github.com/WordPress/ai/pull/1035

Thanks
